sopel/config/core_section.py

```diff
@@ -1048,6 +1048,16 @@ def homedir(self):
     silently from the triggering IRC user's perspective.
     """
 
+    sasl_cert_file = FilenameAttribute('sasl_cert_file')
+    """Filesystem path to a certificate file for CertFP.
+
+    This is expected to be a ``.pem`` file containing both the certificate and
+    private key. Most networks that support CertFP will give instructions for
```

Comment:
Is there a reason to not just provide an example command here? e.g. "`openssl req -new -newkey rsa:4096 -days 3650 -nodes -x509 -sha256 -out sopel.pem -keyout sopel.pem` and accept the defaults"
I guess the services setup instructions may differ?

Reply:
I figure it's safest to force users to look at the documentation from their network, so they 1) use the exact `openssl` command that network wants in case of weird algo/bitness requirements or maximum validity time, etc. and 2) will (more likely) see the instructions for registering the cert fingerprint with services.

Reply:
In my opinion it's best to let the service defines how to generate this file. We are not security expert, and the best we can do is to remind the bot's owner to read the doc of the service they are connecting to.

Reply:
Eugh. That's fair given the downright depressing state of updates/security on most networks, but I'd still kind of like to provide a reasonable default. "You can generate a certificate with `...`, but you should check your network's instructions for setup instructions and any special certificate requirements", since many networks won't have good documentation, or it'll be unnecessarily insecure, generate separate key/cert files, etc.
RSA4096 is secure and universal, SHA256 is secure and... widespread ☹️, and I can't imagine a reason for requiring short expiry of self-signed self-disenrolled difficult-to-automate certificates.

Reply:
I also don't want to make it too easy for users who have no idea what they're doing (artist's rendition of one below) to make use of this feature. The ones who know what's up will have already visited their network's CertFP instructions to set it up for themselves, likely before checking to see if their bot (Sopel) supports it—perhaps even already generated a certificate, and just need to know how to use it.

<details>
<summary>Artist's rendition</summary>
<img src="https://user-images.githubusercontent.com/164140/121473185-e1bd9c80-c987-11eb-89bc-bae6d576993d.png" alt="clueless user">
</details>